### ISSUE TITLE
train/test split is inefficient

### ISSUE THREAD

**Issue by mortonjt** [biocore/rhapsody#44]
This function is currently converting sparse bioms to a dense representation.
https://github.com/biocore/rhapsody/blob/master/rhapsody/util.py#L111

This maybe problematic when scaling these algorithms to very large datasets.

**Comment by fedarko** [comment]
I think this is related to https://github.com/biocore/biom-format/issues/808—until recently, the DataFrame produced by `biom.Table.to_dataframe()` was effectively dense. This caused some silly scaling problems with rankratioviz, also.

For rankratioviz, what I ended up doing to address this problem was extracting the `scipy.sparse.csr_matrix` data from the BIOM table and using that to construct an actually-sparse DataFrame. [Here's the code that does this](https://github.com/fedarko/rankratioviz/blob/88c7c6fc8ed19e25c560b549cec02f23f35c7330/rankratioviz/generate.py#L134) (it's only ~4 lines of work between creating the DF and then populating its IDs). This should work even on BIOM versions with the https://github.com/biocore/biom-format/issues/808 bug.

**Comment by mortonjt** [comment]
Yes! Having the train/test split only applied to the biom tables would be a
huge plus. That's another item on the todo list...

**Comment by nbokulich** [comment]
what about only performing train/test split on the metadata? And then you subsample the biom tables accordingly. This is what is done in q2-sample-classifier, and could be applied here.

**Comment by mortonjt** [comment]
@nbokulich that makes a lot of sense.  And the refactored solution will ultimately require a solution something like that.  Furthermore, getting this right will ultimately help with scalability.

Currently looking into interleaving this with pytorch's [DataLoader class](https://pytorch.org/docs/stable/data.html) - this will help parallel processing so that the GPUs will be saturated (should speed things up at least 2x fold).
